### PR TITLE
Fix `sb3_contrib/ars/policies.py` type hint

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

Requires https://github.com/DLR-RM/stable-baselines3/pull/1188 to be merged

## Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md))

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] The functionality/performance matches that of the source (**required** for new training algorithms or training-related features).
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have included an example of using the feature (*required for new features*).
- [ ] I have included baseline results (**required** for new training algorithms or training-related features).
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the changelog accordingly (**required**).
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by araffin** [comment]
> Requires https://github.com/DLR-RM/stable-baselines3/pull/1188 to be merged

and to be released too

File: sb3_contrib/ars/policies.py [hunk 1]
```diff
@@ -98,17 +102,7 @@ def __init__(
...
-        super().__init__(observation_space, action_space, squash_output=squash_output)
-
-        if isinstance(action_space, gym.spaces.Box):
-            action_dim = get_action_dim(action_space)
-            self.action_net = nn.Linear(self.features_dim, action_dim, bias=with_bias)
-            if squash_output:
-                self.action_net = nn.Sequential(self.action_net, nn.Tanh())
-        elif isinstance(action_space, gym.spaces.Discrete):
-            self.action_net = nn.Linear(self.features_dim, action_space.n, bias=with_bias)
-        else:
```
Comment (araffin) [inline sb3_contrib/ars/policies.py:110]: I guess we handle that in the `supported_action_spaces` parameter in the algorithm and `get_action_dim` won't work in the case of discrete actions

Comment (qgallouedec) [reply sb3_contrib/ars/policies.py:110]: Actually it is handled in the super class.
Or do you mean, having a dedicated supported action space list that would be different from ARSPolicy?

**Comment by araffin** [comment]
One thing that need to be checked before merging: does it break pre-trained models? (we might have to do some manual renaming in case it does)

**Comment by qgallouedec** [comment]
All `rl-baselines3-zoo/tests/test_trained_agents` passed, except `CartPole-v1`. I'm investigating.

**Comment by qgallouedec** [comment]
Probably because `ARSLinearPolicy.action_net` is turned from 
```
Linear(in_features=4, out_features=2, bias=False)
```
to

```
Sequential(
  (0): Linear(in_features=4, out_features=2, bias=False)
)
```

**Comment by qgallouedec** [comment]
```python
from sb3_contrib import ARS
from torch import nn

FOLDER = "rl-trained-agents"
env_id = "CartPole-v1"

for model_name in ["best_model", env_id]:
    model = ARS.load(f"{FOLDER}/ars/{env_id}_1/{model_name}.zip")
    model.policy.action_net = nn.Sequential(model.policy.action_net)
    model.save(f"{FOLDER}/ars/{env_id}_1/{model_name}.zip")
```

It seems to have worked. Does that look right to you?

**Comment by qgallouedec** [comment]
For HF, I think this should also work, but I'm not enough familiar with HF Hub to be sure:

```python
from huggingface_sb3 import load_from_hub, push_to_hub
from torch import nn
from sb3_contrib import ARS

checkpoint = load_from_hub(
    repo_id="sb3/ars-CartPole-v1",
    filename="ars-CartPole-v1.zip",
)
model = ARS.load(checkpoint)
model.policy.action_net = nn.Sequential(model.policy.action_net)

push_to_hub(
    repo_id="sb3/ars-CartPole-v1",
    filename="ars-CartPole-v1.zip",
    commit_message="Update action_net structure",
)
```

**Comment by araffin** [comment]
> All rl-baselines3-zoo/tests/test_trained_agents passed, except CartPole-v1. I'm investigating.

I fixed that in my latest commit ;) (the name of the commit is wrong though ^^ read "load" instead of "save")

**Comment by qgallouedec** [comment]
Look good, I just turned the docstring of `ARS.set_parameters` into a comment, so that the true docstring is inherited from `BaseAlgorithm`.

**Pull request merged**